src/core/Layers.js

```diff
@@ -3,6 +3,7 @@ class Layers {
 	constructor() {
 
 		this.mask = 1 | 0;
+		this.recursive = false;
```

Comment:
Maybe the default should be configurable with a static property?

Reply:
I think the property has to be particularly notable and value contentious to be worth reconfiguring.

Ideally, this would work in user-land for cases like https://github.com/mrdoob/three.js/issues/18937#issuecomment-602252153:

```js
THREE.Layers.prototype.recursive = true;

console.log( new THREE.Layers().recursive ) // false
```

Reply:
In that case, I'd propose changing the default to `true`.

Reply:
According to https://github.com/mrdoob/three.js/pull/19012#issuecomment-613777171, we want to keep `false` as the default.

This is also my expected behavior of layers, tbh.

Reply:
If this isn't even satisfactory for @gkjohnson, then I can simply retract this PR. I'm not aware of how this is handled in user-land today.

Reply:
TBH, I never use layers recursively and if so I would probably traverse through the object hierarchy once and copy the layers to all objects that require the same mask. Granted a property that makes things configurable is easier to use but the PR also proves an increase in complexity in _all_ components checking layers.

I'm not sure how common it is to user layers recursively to to justify this complexity. I'm also not aware common it is to mix recursive layer usage with non-recursive usage. If this is more a global setting then I agree a static `DEFAULT_RECURSIVE` makes sense.

However, I would really like to hear @mrdoob on this change and also on the additional checks in all renderers and raycaster.

Reply:
> If this isn't even satisfactory for @gkjohnson,

I think I'm too far removed from that PR since it was over 4 year ago, now, and haven't been using layers recently. I'll  have to leave this to others.